### ISSUE TITLE
BUG: Autosummary links broken in master

### ISSUE THREAD

**Issue by larsoner** [sphinx-doc/sphinx#7570]
As of commit e5192ba48b45576e636e7dce82ad9183051443ed (according to `git bisect`) sphinx-gallery builds of a [simple test](https://github.com/sphinx-gallery/sphinx-gallery/tree/master/sphinx_gallery/tests/tinybuild) are failing where references to objects are no longer usable:
```
.../tinybuild/auto_examples/future/plot_future_imports.rst:32: WARNING: py:func reference target not found: sphinx_gallery.backreferences.NameFinder
```
I wondered if it was due to #7549, but we explicitly have `autosummary_generate = True` [in the conf.py](https://github.com/sphinx-gallery/sphinx-gallery/blob/master/sphinx_gallery/tests/tinybuild/conf.py#L27) so I don't think so.

I'll keep working on boiling this down to a more minimal example. But if anyone wants to try in the meantime, something like this should work:
```
git clone https://github.com/sphinx-gallery/sphinx-gallery.git
cd sphinx-gallery
pip install -ve .
cd sphinx_gallery/tests/tinybuild
make
```

**Comment by larsoner** [comment]
I have reconfirmed that e5192ba48b45576e636e7dce82ad9183051443ed causes the bug. Before that commit I get no warnings and:

![Screenshot from 2020-04-27 11-01-04](https://user-images.githubusercontent.com/2365790/80387480-b0af0500-8876-11ea-8b98-01a76fe425f9.png)

After that commit:

![Screenshot from 2020-04-27 11-03-37](https://user-images.githubusercontent.com/2365790/80387553-c58b9880-8876-11ea-80b4-8939a1997f3c.png)

Here is the minimal example, just unzip and run `make`:

[tinybuild.zip](https://github.com/sphinx-doc/sphinx/files/4540401/tinybuild.zip)

**Comment by tk0miya** [comment]
Thank you for letting me know. This must be a bug!

**Comment by tk0miya** [comment]
Fixed by #7578
Thank you for reporting.
